packages/GisidaLite/src/tests/index.test.tsx

```diff
@@ -1,11 +1,56 @@
-import { shallow } from 'enzyme';
+import { mount, shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
 import React from 'react';
-import { arePropsEqual, gisidaLiteDefaultProps, GisidaLiteProps, MemoizedGisidaLite } from '../';
+import ReactMapboxGl from 'react-mapbox-gl';
+import {
+  arePropsEqual,
+  gisidaLiteDefaultProps,
+  GisidaLiteProps,
+  MemoizedGisidaLite,
+  ReactMapboxGlProps
+} from '..';
+
+jest.mock('react-mapbox-gl');
 
 describe('component/GisidaLite', () => {
   it('renders without crashing', () => {
     shallow(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
   });
+  it('renders props correctly', () => {
+    const wrapper = mount(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
+    expect(wrapper.props()).toMatchSnapshot('gisida lite props');
+  });
+  it('renders component correctly', () => {
+    const wrapper = mount(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
+    expect(toJson(wrapper)).toMatchSnapshot('gisida lite render');
+  });
+  it('reacts to props changes accordingly in relation to ReactMapboxGl constructor', () => {
+    const updatedReactMapboxglProps = {
+      accessToken: '',
+      attributionControl: true,
+      customAttribution: '',
+      injectCSS: true
+    };
+    const mockReactMapboxGlMock = jest.fn();
+    (ReactMapboxGl as jest.Mock).mockImplementationOnce(() => mockReactMapboxGlMock);
+
+    const wrapper = mount(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
+    expect(mockReactMapboxGlMock).toHaveBeenCalledTimes(1);
+    expect(wrapper.props().reactMapboxGlMapFactoryUtilConfigs).toEqual(ReactMapboxGlProps);
+    // should not re-instanciate reactmapboxgl on other props change
+    wrapper.setProps({
+      ...gisidaLiteDefaultProps,
+      layers: []
+    });
+    expect(mockReactMapboxGlMock).toHaveBeenCalledTimes(1);
+    expect(wrapper.props().reactMapboxGlMapFactoryUtilConfigs).toEqual(ReactMapboxGlProps);
+    // should re-instanciate map on reactmapboxgl factory options change
+    wrapper.setProps({
+      ...gisidaLiteDefaultProps,
```

Comment:
Will the map re-render if props such as layers change?  Sorry but that is not clear here,

Reply:
A change in reactMapboxGlMapFactoryUtilConfigs prop is the only way to get the map re-instanciated.

Reply:
@kahummer is that not going to be a problem?  I mean a lot of the use case is that someone wants to update layers and see a change on the map - ama?